secondary-server-1/src/main/java/com/replicated_log/secondary_server_1/service/ItemServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.replicated_log.secondary_server_1.service;
+
+import com.replicated_log.secondary_server_1.model.Item;
+import com.replicated_log.secondary_server_1.repository.ItemRepository;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.TreeSet;
+
+@Service
+public class ItemServiceImpl implements ItemService {
+
+    private final Logger LOG = LogManager.getLogger(ItemServiceImpl.class);
+
+    private final Integer FIRST_ID_OF_DEMO_ITEMS = 1;
+
+    @Autowired
+    private ItemRepository itemRepository;
+
+    @Override
+    public boolean addItem(Item item) {
+        LOG.info("S--> ItemServiceImpl: Adding item " + item);
+        return itemRepository.append(item);
+    }
+
+    @Override
+    public Set<Item> getItems() {
+        return itemRepository.findAll();
+    }
+
+    @Override
+    public Set<Item> getItemsForDemo() {
```

Comment:
the same , what is the difference between getItems and getItemsForDemo

Reply:
findAllItemsForDemo() method shows messages from Secondarys storage in 'total order' according to Projects Requirements:
If secondary has received messages [msg1, msg2, msg4], it shouldn’t display the message ‘msg4’ until the ‘msg3’ will be received

findAllItems() method shows all excisting messages from Secondary`s storage